src/test/java/org/javarosa/core/test/Scenario.java

```diff
@@ -119,6 +118,84 @@ private Scenario(FormDef formDef, FormEntryController formEntryController) {
         this.formEntryController = formEntryController;
     }
 
+    // region Miscelaneous
+
+    public static TreeReference getRef(String xpath) {
+        try {
+            TreeReference reference = ((XPathPathExpr) XPathParseTool.parseXPath(xpath)).getReference();
+            for (int i = 0; i < reference.size(); i++)
+                if (reference.getPredicate(i) != null
+                    && !reference.getPredicate(i).isEmpty()
+                    && reference.getPredicate(i).get(0) instanceof XPathNumericLiteral
+                ) {
+                    int multiplicity = Double.valueOf(((XPathNumericLiteral) reference.getPredicate(i).get(0)).d).intValue();
+                    reference.setMultiplicity(i, multiplicity);
+                    reference = reference.removePredicates(i);
+                }
+            return reference;
+        } catch (XPathSyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void newInstance() {
+        formDef.initialize(true, new InstanceInitializationFactory());
+    }
+
+    public void setLanguage(String language) {
+        formEntryController.setLanguage(language);
+    }
+
+    public EvaluationContext getEvaluationContext() {
+        return formDef.getEvaluationContext();
+    }
+
+    public Scenario populateDynamicChoices() {
+        FormIndex backupIndex = formEntryController.getModel().getFormIndex();
+        while (!atTheEndOfForm()) {
+            silentNext();
```

Comment:
This populates dynamic questions of the next question? Why? Doesn't getting the choices implicitly populate them if dynamic? This seems like it would better match what an actual client does.

Reply:
Nevermind this method. This was something that I tried to simplify the `SelectChoice` tests but then I decided to go in another direction. I committed this method by mistake :/

Tests can call `Scenario.choicesOf` as always to get select choices, which populates them implicitly.

My original idea was to make it explicit at critical points of a test for future readers of tests, but ultimately I thought it wasn't bringing so much value to the table. The comments that are already in place work OK, and, as you've said, it is closed to what clients would do.